src/js/editor/editor.js

```diff
@@ -157,10 +155,11 @@ function bindKeyListeners(editor) {
       editor.handleNewline(event);
     } else if (key.isPrintable()) {
       if (editor.cursor.hasSelection()) {
-        let result = editor.run((postEditor) => {
-          return postEditor.deleteRange(editor.cursor.offsets);
+        let offsets = editor.cursor.offsets;
+        editor.run((postEditor) => {
+          postEditor.deleteRange(editor.cursor.offsets);
         });
-        editor.cursor.moveToMarker(result.currentMarker, result.currentOffset);
+        editor.cursor.moveToSection(offsets.headSection, offsets.headSectionOffset);
```

Comment:
We should just add a `restoreOffset` that takes the offset. It would need to error when the expected sections were no longer present, but would be really useful besides then.